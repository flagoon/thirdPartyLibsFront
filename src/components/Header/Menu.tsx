import React from 'react';
import styled from 'styled-components';

interface Button {
  color?: string;
}

const Menu = (): JSX.Element => {
  return (
    <div>
      <MenuContainer>
        <MenuButton color="red" onClick={(): void => console.log('Clicked1')}>
          +
        </MenuButton>
        <MenuButton color="blue" onClick={(): void => console.log('Clicked2')}>
          Add multiple
        </MenuButton>
        <MenuButton>
          <label htmlFor="filter">
            Filter:{' '}
            <input
              name="filter"
              id="filter"
              data-test="filter"
              placeholder="filter..."
              onChange={(e): void => console.log(e.target)}
            />
          </label>
        </MenuButton>
      </MenuContainer>
    </div>
  );
};
export default Menu;

export const MenuContainer = styled.ul`
  display: flex;
  align-items: center;
  margin: 0;
  padding: 0;
`;

export const MenuButton = styled.li`
  color: ${(props: Button): string => (props.color ? props.color : 'black')};
  list-style-type: none;
  cursor: pointer;

  &:not(:first-child) {
    margin-left: 1rem;
    padding: 0.5rem;
  }
`;
