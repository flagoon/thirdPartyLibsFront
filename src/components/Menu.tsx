import React, { useContext } from 'react';
import { Grid } from '@material-ui/core';
import FilterContext from '../Providers/Filter/FilterContext';

export const Menu = (): JSX.Element => {
  const { filterValue, changeFilterValue } = useContext(FilterContext);
  return (
    <Grid container item direction="row" alignItems="center">
      <input
        placeholder="filter..."
        onChange={changeFilterValue}
        value={filterValue}
      />
      {filterValue !== '' ? <div onClick={changeFilterValue}>x</div> : null}
    </Grid>
  );
};
