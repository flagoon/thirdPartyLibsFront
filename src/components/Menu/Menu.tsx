import { Grid } from '@material-ui/core';
import React from 'react';
import Filter from './Filter/Filter';

function Menu(): JSX.Element {
  return (
    <Grid container item direction="row" alignItems="center">
      <Filter />
    </Grid>
  );
}

export default Menu;
