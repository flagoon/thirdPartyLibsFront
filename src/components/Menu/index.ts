import Filter from './Filter/Filter';
import Menu from './Menu';

export { Filter, Menu };
