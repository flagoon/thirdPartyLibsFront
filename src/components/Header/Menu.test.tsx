import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Menu from './Menu';

describe('Menu', () => {
  it('should match the snapshot', () => {
    const openModal = jest.fn();
    const { asFragment } = render(<Menu onVisibilityChange={openModal} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render plus button and filter', () => {
    const openModal = jest.fn();
    const { getByTestId } = render(<Menu onVisibilityChange={openModal} />);
    expect(getByTestId('plusIcon')).toBeInTheDocument();
  });
  it('should call onVisibilityChange after click', () => {
    const openModal = jest.fn();
    const { getByTestId } = render(<Menu onVisibilityChange={openModal} />);
    fireEvent.click(getByTestId('plusIcon'));
    expect(openModal).toHaveBeenCalledTimes(1);
  });
});
