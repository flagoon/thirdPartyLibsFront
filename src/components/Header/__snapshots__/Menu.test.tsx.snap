// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-align-items-xs-center"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root sc-AxjAm gJQWcM"
      data-testid="plusIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
      />
    </svg>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline"
    >
      <input
        class="MuiInputBase-input MuiInput-input"
        id="filter"
        name="filter"
        placeholder="filter..."
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
